kolibri/core/content/utils/content_request.py

```diff
@@ -195,6 +198,18 @@ def incomplete_downloads_queryset():
     )
 
 
+def _get_content_storage_space():
+    """
+    Calculates and returns the total space used for content storage
+    """
+    size = 0
+    for path, dirs, files in os.walk(OPTIONS["Paths"]["CONTENT_DIR"]):
```

Comment:
This is calculating the total disk space used for all content downloaded, whereas the setting is specifically for limiting the space used by automatic and learner initiated downloads (it looks like maybe the [issue](https://github.com/learningequality/kolibri/issues/10385) didn't specify this, but you can see from the wording in the screenshot for the setting).

This would mean that if someone had pre-provisioned a device with 1GB of resources via the admin panel, and then set the limit to 250MB, then they would never be able to download anything.

To accomplish this, it would be better to infer this from the completed DownloadRequests currently on the device, which should also include the total size.

Reply:
Okay, I see what you mean. So should I instead do a query similar to what is being done [here](https://github.com/learningequality/kolibri/blob/8153642723924bdff7066bbb69e9da293662942f/kolibri/core/content/utils/content_request.py#L184:~:text=def%20incomplete_downloads_queryset()%3A), but for DownloadRequests with a status of completed and return the total size of downloaded files from the completed DownloadRequests instead?

Reply:
Yes, that seems right - and then summing all the total_sizes from there.